src/pks/mpc/mpc_coupled_dualmedia_water.cc

```diff
@@ -0,0 +1,157 @@
+#include "Teuchos_XMLParameterListHelpers.hpp"
+#include "EpetraExt_RowMatrixOut.h"
+
+#include "mpc_coupled_dualmedia_water.hh"
+
+namespace Amanzi {
+
+MPCCoupledDualMediaWater::MPCCoupledDualMediaWater(Teuchos::ParameterList& FElist,
+                  const Teuchos::RCP<Teuchos::ParameterList>& plist,
+                  const Teuchos::RCP<State>& S,
+                  const Teuchos::RCP<TreeVector>& soln) :
+    PK(FElist, plist,  S, soln),
+    StrongMPC<PK_BDF_Default>(FElist, plist,  S, soln) {
+
+}
+
+
+
+void  MPCCoupledDualMediaWater::Setup(const Teuchos::Ptr<State>& S) {
+   // tweak the sub-PK parameter lists
+  StrongMPC<PK_BDF_Default>::Setup(S);
+  
+  std::cout<<"Setup\n";
+  Teuchos::Array<std::string> names = plist_->get< Teuchos::Array<std::string> >("PKs order");
+  int npks = names.size();
+  for (int i=0; i!=npks; ++i) {
+    Teuchos::RCP<const CompositeVectorSpace> tmp = solution_->Map().SubVector(i)->Data();
+    std::cout<<names[i]<<" "<<tmp<<"\n";
+  }
+
+  pks_list_->sublist(names[1]).set("coupled to subsurface via head", true);
+    // cast the PKs
+  integrated_flow_pk_ = Teuchos::rcp_dynamic_cast<StrongMPC<PK_PhysicalBDF_Default> >(sub_pks_[0]);
+  AMANZI_ASSERT(integrated_flow_pk_ != Teuchos::null);
+
+  macro_flow_pk_ = sub_pks_[1];
+  matrix_flow_pk_ = integrated_flow_pk_->get_subpk(0);
+  surf_flow_pk_ = integrated_flow_pk_->get_subpk(1);
+
+  macro_flux_key_ = "macropore-mass_flux";
```

Comment:
Please don't hard-code keys -- use readKey.  This allows these PKs to be used more generally.